src/matcha_ml/config/matcha_config.py

```diff
@@ -0,0 +1,174 @@
+                f"The component with the name '{component_name}' could not be found."
+            )
+
+        return component
+
+    def to_dict(self) -> Dict[str, Dict[str, str]]:
+        """A function to convert the MatchaConfig class into a dictionary.
+
+        Returns:
+            Dict[str, Dict[str, str]]: the MatchaState as a dictionary.
+        """
+        state_dictionary = {}
+        for config_component in self.components:
+            state_dictionary[config_component.name] = {
+                property.name: property.value
+                for property in config_component.properties
+            }
+
+        return state_dictionary
+
+    @staticmethod
+    def from_dict(state_dict: Dict[str, Dict[str, str]]) -> "MatchaConfig":
+        """A function to convert a dictionary representation of the Matcha config file into a MatchaConfig instance.
+
+        Args:
+            state_dict (Dict[str, Dict[str, str]]): the dictionary representation of the Matcha config file.
+
+        Returns:
+            MatchaConfig: the MatchaConfig representation of the MatchaConfig instance.
+        """
+        components: List[MatchaConfigComponent] = []
+        for resource, properties in state_dict.items():
+            components.append(
+                MatchaConfigComponent(
+                    name=resource,
+                    properties=[
+                        MatchaConfigComponentProperty(name=key, value=value)
+                        for key, value in properties.items()
+                    ],
+                )
+            )
+
+        return MatchaConfig(components=components)
+
+
+class MatchaConfigService:
+    """A service for handling the Matcha config file."""
+
+    @staticmethod
+    def write_matcha_config(matcha_config: MatchaConfig) -> None:
+        """A function for writing the local Matcha config file.
+
+        Args:
+            matcha_config (MatchaConfig): the MatchaConfig representation of the MatchaConfig instance.
+        """
+        local_config_file = os.path.join(os.getcwd(), DEFAULT_CONFIG_NAME)
+
+        with open(local_config_file, "w") as file:
+            json.dump(matcha_config.to_dict(), file)
+
+    @staticmethod
+    def read_matcha_config() -> MatchaConfig:
+        """A function for reading the Matcha config file into a MatchaConfig object.
+
+        Returns:
+           MatchaConfig: the MatchaConfig representation of the MatchaConfig instance.
+        """
+        local_config_file = os.path.join(os.getcwd(), DEFAULT_CONFIG_NAME)
+
+        if os.path.exists(local_config_file):
+            with open(local_config_file) as config:
+                local_config = json.load(config)
+
+            return MatchaConfig.from_dict(local_config)
+        else:
+            raise MatchaError(
+                "No 'matcha.config.json' file found, please generate one by running 'matcha provision', or add an existing 'matcha.config.json' file to the root project directory."
```

Comment:
Nit: 

Make use of dynamic formatting here: `f"No {DEFAULT_CONFIG_NAME} file found, ..."`

Reply:
Updated.